scripts/install/functions.sh

```diff
@@ -1085,3 +1087,125 @@ function compareSemver() {
+   
+    if [ ! -f "$SPARTACUS_FEATURES_MODULE_PATH" ]; then
+        echo "Incorrect path: $SPARTACUS_FEATURES_MODULE_PATH"
+        return
+    fi
+
+    if [ "$JDK_VERSION" = "JDK21" ]; then
+        authorizationCodeFlowByDefault_value="true"
+    fi
+
+    if ! grep -q 'authorizationCodeFlowByDefault' "$SPARTACUS_FEATURES_MODULE_PATH"; then
+        echo "authorizationCodeFlowByDefault toggle not found in file"
+        return
+    fi
+
+    if grep -E "\"?authorizationCodeFlowByDefault\"?:[[:space:]]*$authorizationCodeFlowByDefault_value" "$SPARTACUS_FEATURES_MODULE_PATH" > /dev/null; then
+      echo "authorizationCodeFlowByDefault is already set with $authorizationCodeFlowByDefault_value, no change needed."
+    else
+      sed_inplace -E "s/(\"?authorizationCodeFlowByDefault\"?:[[:space:]]*)(true|false)/\1$authorizationCodeFlowByDefault_value/" "$SPARTACUS_FEATURES_MODULE_PATH"
+        if grep -E "\"?authorizationCodeFlowByDefault\"?:[[:space:]]*$authorizationCodeFlowByDefault_value" "$SPARTACUS_FEATURES_MODULE_PATH" > /dev/null; then
+            echo "authorizationCodeFlowByDefault has been successfully updated to $authorizationCodeFlowByDefault_value with JDK version: $JDK_VERSION"
+        else
+            echo "Failed to update authorizationCodeFlowByDefault with JDK version: $JDK_VERSION"
+        fi
+    fi
+ 
+    echo "*********** spartacus-features.module.ts ***********"
+    echo $SPARTACUS_FEATURES_MODULE_PATH
+    cat "$SPARTACUS_FEATURES_MODULE_PATH"
+    echo "*********** EOF spartacus-features.module.ts ***********"
+
+   
+     if [ "$JDK_VERSION" = "JDK21" ]; then
+        addAuthConfig "$app_dir"
+     fi
+}
+
+# Adds AuthConfig to the Spartacus configuration module.
+# Use case: B2C and B2B are hosted on separate domains, requiring two OAuth clients in backoffice.
+# The defaultAuthConfig must be overwritten to match the backoffice OAuth client configuration.
+function addAuthConfig {
+    if [ -z "$AUTH_CONFIG" ]; then
+        echo "AUTH_CONFIG is null or empty, exiting function."
+        return
+    fi
+
+    local app_dir="${1}"
+    SPARTACUS_CONFIGURATION_MODULE_PATH="${INSTALLATION_DIR}/${app_dir}/src/app/spartacus/spartacus-configuration.module.ts"
+    if [ ! -f "$SPARTACUS_CONFIGURATION_MODULE_PATH" ]; then
+        echo "Incorrect path: $SPARTACUS_CONFIGURATION_MODULE_PATH"
+        return
+    fi
+
+    clean_module_file() {
+       #Remove all empty lines then add one above @NgModule
+       echo "Cleaning up empty lines"
+       sed_inplace -e '/^[[:space:]]*$/d' -e '/@NgModule/i\' "$SPARTACUS_CONFIGURATION_MODULE_PATH"
+    }
+
+    if [ "$ADD_AUTH_CONFIG" = false ]; then
+        echo "ADD_AUTH_CONFIG is false -> Do not add AuthConfig."
+        echo "Remove authconfig import and provider if it was previously added."
+        perl -0777 -pi -e 's/provideConfig\(<AuthConfig>.*?\}\),\s*//gs' "$SPARTACUS_CONFIGURATION_MODULE_PATH"
```

Comment:
use case: JDK21 without adding authConfig (ADD_AUTH_CONFIG=false).
the script still open and clean configuration.module.t file.
Reason is that user might have previously run the app with custom authConfig (AUTH_CONFIG with ADD_AUTH_CONFIG=true). we thus need a mechanism to clean previous config even if it is the first attempt.

To avoid this cleaning mechansim, the user can comment the AUTH_CONFIG var from config.default.sh, the function will exit on line 1151.